include/envoy/network/io_handle.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * IoHandle: an abstract interface for all I/O operations
+ */
+class IoHandle {
+public:
+  IoHandle() {}
+
+  virtual ~IoHandle() {}
+
+  // TODO(sbelair2) remove fd() method
+  virtual int fd() const PURE;
+
+  virtual void close() PURE;
```

Comment:
Comment with full javadoc style comments, please.
It'd be good to comment fd() as well but that's going away.  This one isn't :-)

Reply:
Done.